### PR TITLE
Record Types in Example.fr

### PR DESCRIPTION
**Pull request by jakubbinter**
Added Record Types in example.fr

### CODE REVIEW

File: Example.fr [hunk 1]
```diff
@@ -5,6 +5,20 @@ module Example;
 
 import System.Console;
 
+type BasicMath(num1:int32, num2:int32);
```
Comment (WhiteBlackGoose) [inline Example.fr:8]: Can we have space after `:` please? We haven't decided about code style yet, but let's keep it as `name: Type`

File: Example.fr [hunk 2]
```diff
@@ -5,6 +5,20 @@ module Example;
...
+
+impl BasicMath
+{
+    func Add(this):int32
```
Comment (WhiteBlackGoose) [inline Example.fr:12]: For now, methods should be `camelCase` (or snake_case?). anyway, this should be `add`

Also, space before type

File: Example.fr [hunk 3]
```diff
@@ -5,6 +5,20 @@ module Example;
...
+    {
```
Comment (WhiteBlackGoose) [inline Example.fr:13]: `{` comes in the previous line

File: Example.fr [hunk 4]
```diff
@@ -5,6 +5,20 @@ module Example;
...
+        return this.num1+this.num2;
```
Comment (WhiteBlackGoose) [inline Example.fr:14]: Spaces before and after `+`

File: Example.fr [hunk 5]
```diff
@@ -5,6 +5,20 @@ module Example;
...
+    }
+    func Sub(this):int32
```
Comment (WhiteBlackGoose) [inline Example.fr:16]: Naming, spacing. Also let's do a new line between the two methods

File: Example.fr [hunk 6]
```diff
@@ -5,6 +5,20 @@ module Example;
...
+    {
```
Comment (WhiteBlackGoose) [inline Example.fr:17]: Should come in the previous line

File: Example.fr [hunk 7]
```diff
@@ -5,6 +5,20 @@ module Example;
...
+        return this.num1-this.num2;
```
Comment (WhiteBlackGoose) [inline Example.fr:18]: Spacing

File: Example.fr [hunk 8]
```diff
@@ -28,5 +42,8 @@ func main(): int32 {
     Console.WriteLine(s);
     val f = distance(5.5, 5.0, 10.6, 5.5);
     
+    val math=BasicMath(2,3);
```
Comment (WhiteBlackGoose) [inline Example.fr:45]: Spacing (also after comma)

File: Example.fr [hunk 9]
```diff
@@ -28,5 +42,8 @@ func main(): int32 {
...
+    Console.Writeline("2+3="math.Add());
+    Console.Writeline("2-3="math.Sub());
```
Comment (WhiteBlackGoose) [inline Example.fr:47]: This doesn't look valid. Did you mean
```
"2 + 3 = ${math.add()}"
```

Comment (jakubbinter) [reply Example.fr:47]: Crap forgot to put +

File: Example.fr (hunk 3 above)
Comment (jakubbinter) [reply Example.fr:13]: Don't like the { on same line as methon name bit i will change it

File: Example.fr [hunk 10]
```diff
@@ -5,6 +5,18 @@ module Example;
 
 import System.Console;
 
+type BasicMath(num1: int32, num2: int32);
+
+impl BasicMath{
```
Comment (WhiteBlackGoose) [inline Example.fr:10]:
```suggestion
impl BasicMath {
```

**Review by WhiteBlackGoose — APPROVED** [review]

File: Example.fr [hunk 11]
```diff
@@ -5,6 +5,26 @@ module Example;
 
 import System.Console;
 
+type Fraction(numerator: int32, denominator: int32);
+
+impl Fraction {
+    func multiply(this, other: Frcation): Fraction {
```
Comment (WhiteBlackGoose) [inline Example.fr:11]: Typo

File: Example.fr [hunk 12]
```diff
@@ -5,6 +5,26 @@ module Example;
...
+        val numerator = this.numerator * other.numerator;
+        val denominator = this.denominator * other.denominator;
+        return Fraction(numerator, denominator);
+    }
+    
+    func divide(this, other: Frcation): Fraction {
```
Comment (WhiteBlackGoose) [inline Example.fr:17]: Typo

File: Example.fr [hunk 13]
```diff
@@ -28,5 +48,12 @@ func main(): int32 {
     Console.WriteLine(s);
     val f = distance(5.5, 5.0, 10.6, 5.5);
     
+    val fractionTwoThirds = Fraction(2, 3);
+    val fractionOneFifth = Fraction(1, 5);
+    val fractionMultiplied = fractionTwoThirds.multiply(fractionOneFifth); // 2/15
+    val fractionDivided = fractionTwoThirds.divide(fractionOneFifth); // 10/3
```
Comment (WhiteBlackGoose) [inline Example.fr:54]: Can we make it a bit conciser? The names are huge

File: Example.fr [hunk 14]
```diff
@@ -28,5 +48,12 @@ func main(): int32 {
...
+    Console.WriteLine("Multiplied: ${fractionMultiplied.ToString()}");
+    Console.WriteLine("Divided: ${fractionDivided.ToString()}");
```
Comment (WhiteBlackGoose) [inline Example.fr:56]: You don't need `ToString()` when it's string interpolation

**Review by WhiteBlackGoose — CHANGES_REQUESTED** [review]

File: Example.fr [hunk 15]
```diff
@@ -5,6 +5,26 @@ module Example;
...
+        return Fraction(numerator, denominator);
+    }
+    
+    func divide(this, other: Frcation): Fraction {
+        val numerator = this.numerator * other.denominator;
+        val denominator = this.denominator * other.numerator;
+        return Fraction(numerator, denominator);
+    }
+    
+    func ToString(this): _ {
```
[the remark this answers is not in the document]
Comment (jakubbinter) [reply Example.fr:23]: Yea was not sure how it eill be i left it like this for now, we should disccuss this on the server

File: Example.fr (hunk 14 above)
Comment (WhiteBlackGoose) [reply Example.fr:56]: You don't need a call there, it's called automatically

File: Example.fr [hunk 16]
```diff
@@ -5,6 +5,26 @@ module Example;
...
+    }
+    
+    func divide(this, other: Fraction): Fraction {
+        val numerator = this.numerator * other.denominator;
+        val denominator = this.denominator * other.numerator;
+        return Fraction(numerator, denominator);
+    }
+    
+    func ToString(this): _ {
+        return "${this.numerator} / ${this.denominator}";
```
[the remark this answers is not in the document]
Comment (jakubbinter) [reply Example.fr:24]: Makes sence

**Review by WhiteBlackGoose — APPROVED** [review]

File: Example.fr [hunk 17]
```diff
@@ -28,5 +46,12 @@ func main(): int32 {
     Console.WriteLine(s);
     val f = distance(5.5, 5.0, 10.6, 5.5);
     
+    val frac1 = Fraction(2, 3);
+    val frac2 = Fraction(1, 5);
+    val multiplied = frac1.multiply(frac2); // 2/15
+    val divided = frac1.divide(frac2); // 10/3
+    Console.WriteLine("Multiplied: ${multiplied}");
```
Comment (LPeter1997) [inline Example.fr:53]: Both examples could just use the syntax `$identifier`. Like `$multiplied` and `$divided`. You could inline one of them actually into `${...}` to show that they are for complex expressions

**Review by LPeter1997 — COMMENTED** [review]
Added a nit, lgtm otherwise

File: Example.fr (hunk 17 above)
Comment (jakubbinter) [reply Example.fr:53]: Oh never did anything with kotlin, just did what goose told me

Comment (jakubbinter) [reply Example.fr:53]: > You could inline one of them actually into ${...} to show that they are for complex expressions  

Like 
`Console.WriteLine("Divided: ${frac1.divide(frac2)}");

**Pull request merged**